packages/yew/src/scheduler.rs

```diff
@@ -24,12 +24,18 @@ struct Scheduler {
     destroy: Vec<Box<dyn Runnable>>,
     create: Vec<Box<dyn Runnable>>,
     update: Vec<Box<dyn Runnable>>,
-    render_first: VecDeque<Box<dyn Runnable>>,
-    render: RenderScheduler,
 
-    /// Stacks to ensure child calls are always before parent calls
-    rendered_first: Vec<Box<dyn Runnable>>,
-    rendered: RenderedScheduler,
+    /// A Binary Tree Map here guarantees components with lower id (parent) is rendered first and
+    /// no more than 1 render can be scheduled before a component is rendered.
+    ///
+    /// Parent can destroy child components but not otherwise, we can save unnecessary render by
+    /// rendering parent first.
+    render_first: BTreeMap<usize, Box<dyn Runnable>>,
+    render: BTreeMap<usize, Box<dyn Runnable>>,
+
+    /// Binary Tree Map to guarantee children rendered are always called before parent calls
+    rendered_first: BTreeMap<usize, Box<dyn Runnable>>,
+    rendered: BTreeMap<usize, Box<dyn Runnable>>,
```

Comment:
Compared to Vec, how much more expensive is using a `BTreeMap`?

Reply:
I wouldn't be too concerned about the performance of BTreeMap, due to the following:

1. The original `RenderScheduler` and `RenderedScheduler` are a combination of `HashMap` and `Vec`, not just `VecDeque`.
2. BTreeMap can attain millions of operations with ease.
    > https://github.com/Rufflewind/bench-maps
3. Parent first render saves DOM operations which is significantly more expensive.